Make cluster e2e non docker specific.

Random-Liu opened the pull request:
Fixes https://github.com/kubernetes/kubernetes/issues/57979.

This PR:
1) Install `crictl` for GCE GCI/Ubuntu test cluster.
2) Change cluster e2e test to use `crictl`. If `crictl` is not installed, it will fall back to `docker` CLI.

Please note that, this PR:
1) Only installs `crictl` for GCI/Ubuntu image on GCE. If other image maintainers want to support this, it's very easy to install `crictl` `v1.0.0-alpha.0`:
```console
$ wget -O crictl https://storage.googleapis.com/kubernetes-release/crictl/crictl-v1.0.0-alpha.0-linux-amd64
$ chmod +x crictl
// The sha1sum is `075190a36a03beb08065f279693f3e369e70cf99`.
```
2) Only installs `crictl` for test cluster. We may want to install it by default in production cluster after `crictl` is beta in v1.10.

@yujuhong @feiskyer @mrunalp @mikebrow 
/cc @dchen1107 for approval
@kubernetes/sig-node-pr-reviews @kubernetes/sig-testing-bugs 

Signed-off-by: Lantao Liu <lantaol@google.com>

**Release note**:
<!--  Write your release note:
1. Enter your extended release note in the below block. If the PR requires additional action from users switching to the new release, include the string "action required".
2. If no release note is required, just write "NONE".
-->
```release-note
none
```

Random-Liu commented:
/test pull-kubernetes-e2e-gke-gci
/test pull-kubernetes-unit

feiskyer reviewed (approved):
/lgtm

feiskyer commented:
/assign @mikedanese

mikebrow reviewed (commented):
The code changes for ensuring crictl is available for GCE all look good. I don't see: "Change cluster e2e test to use crictl. If crictl is not installed, it will fall back to docker CLI."

Random-Liu commented:
@mikebrow It's in test/e2e/framework/utils.go

mikebrow commented:
@Random-Liu I only see calls for the ps command (health check) being updated.. I don't see "Change cluster e2e test to use crictl. If crictl is not installed, it will fall back to docker CLI." Do you mean change cluster e2e test to use crictl when health status checking?

Random-Liu commented:
@mikebrow Some e2e test use those util functions, such as [here](https://github.com/kubernetes/kubernetes/blob/dca85f046fad9282637ba211bd60d8b04ad9f72c/test/e2e/storage/persistent_volumes-disruptive.go#L190).

Random-Liu commented:
Discussed with @yujuhong offline. She brought up a very good point.
Today CRI is not versioned yet. Using `crictl` in our test environment means that once we make a breaking change to CRI, the test using `crictl` will fail.

We should avoid that.

Actually for this particular change, we can use `ps` and `kill` directly which is general for all container runtime.

I'll make the change.

Random-Liu committed:
Use linux commands instead of docker commands.

Random-Liu commented:
Remove LGTM for now in favor of https://github.com/kubernetes/kubernetes/pull/58024#issuecomment-356791939. @feiskyer 

Addressed comments. @yujuhong

feiskyer commented:
> Today CRI is not versioned yet. Using crictl in our test environment means that once we make a breaking change to CRI, the test using crictl will fail.

Agreed. LGTM

yujuhong commented on test/e2e/framework/util.go:
```diff
@@ -3900,9 +3900,7 @@ func sshRestartMaster() error {
-		// `kube-apiserver_kube-apiserver` matches the name of the apiserver
-		// container.
-		command = "sudo docker ps | grep kube-apiserver_kube-apiserver | cut -d ' ' -f 1 | xargs sudo docker kill"
+		command = "sudo ps -ef | grep \"[k]ube-apiserver\" | awk '{print $2}' | xargs sudo kill"
```
Why not just use `pidof`?

yujuhong replied:
Also, it might be nice to factor this out as a function. Not required by this PR though.

Random-Liu replied:
Will do.

mikedanese commented:
/approve

feiskyer commented:
/lgtm
